<a href="https://colab.research.google.com/github/renanpbento/python-training/blob/main/IGTI_Mod2_Practice.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**IGTI Bootcamp - Module II: Practical Work**

Goals - practice the following concepts:
- Creation and manipulation of numpy arrays
- DataFrames and Series in Pandas
- Machine Learning fundamentals

Preparing the exercise:

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
# Code 1

Z = np.zeros((4, ))

print("Z", Z)

Z [0. 0. 0. 0.]


Code 1 is a row-matrix composed solely by zeros

In [ ]:
# Code 2

Z[1] = 1

print("Z", Z)

Z [0. 1. 0. 0.]


Code 2 updates the matrix by substituting the second element from "0" to "1"

In [ ]:
# Code 3

Z[1: ] = 1

print("Z", Z)

Z [0. 1. 1. 1.]


Code 3 transforms the row beginning from the second element (index 1)

In [ ]:
# Code 4

Z = np.zeros((4, ))

Z[:3] = 1

print("Z", Z)

Z [1. 1. 1. 0.]


Code 4 updates a row-matrix composed of zeros. There is more than one method to solve this exercise

In [ ]:
# Code 5

X = np.ones((2, 2)) * 2

print("X:\n", X)

X:
 [[2. 2.]
 [2. 2.]]


Code 5 establishes a 2x2 matrix composed of "2s". Like the previous exercise, there is more than one method to solve this

In [ ]:
# Code 6

X = np.array([[1, 2], [3, 4]])
Y = X[0, :]
Y[1] = 10

print("X:\n", X)

X:
 [[ 1 10]
 [ 3  4]]


Code 6 creates a matrix 2x2 and establishes a changing element {1,2} (index 0,1) through Y

In [ ]:
# Code 7

X = np.array([[1,3], [11,10]])

print(np.mean(X[X > np.pi]))

10.5


Code 7 creates a 2x2 matrix and calculates the mean of those elements that are greater than Pi (approx 3.14), in this case 10 and 11:

(10 + 11)/2 = 21/2 = 10.5

In [ ]:
# Code 8

data = {'animal': ['cat', 'cat', 'snake', 'dog', 'dog', 'cat', 'snake', 'cat', 'dog', 'dog'],
        'age': [2.5, 3, 0.5, np.nan, 5, 2, 4.5, np.nan, 7, 3],
        'visits': [ 1, 3, 2, 3, 2, 3, 1, 1, 2, 1],
        'priority': ['yes', 'yes', 'no', 'yes', 'no', 'no', 'no', 'yes', 'no', 'no']}
      
labels = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j']

print(data)

{'animal': ['cat', 'cat', 'snake', 'dog', 'dog', 'cat', 'snake', 'cat', 'dog', 'dog'], 'age': [2.5, 3, 0.5, nan, 5, 2, 4.5, nan, 7, 3], 'visits': [1, 3, 2, 3, 2, 3, 1, 1, 2, 1], 'priority': ['yes', 'yes', 'no', 'yes', 'no', 'no', 'no', 'yes', 'no', 'no']}


Code 8 creates a dictionary. The data could pertain to a Veterinary Clinic/Hospital.

In [ ]:
# Code 9

df = pd.DataFrame(data = data, index = labels)

#df.sort_index(by = 'visits')
df.sort_values(by = 'visits', ascending = False)


animal  age  visits priority
b    cat  3.0       3      yes
d    dog  NaN       3      yes
f    cat  2.0       3       no
c  snake  0.5       2       no
e    dog  5.0       2       no
i    dog  7.0       2       no
a    cat  2.5       1      yes
g  snake  4.5       1       no
h    cat  NaN       1      yes
j    dog  3.0       1       no

Code 9 transforms the dictionary from Code 8 into a Pandas data frame. 

In [ ]:
# Code 10

y_true = np.array([1, 2, 1])
y_pred = np.array([1.1, 1.98, 1.05])

eqm = np.sqrt(((y_true - y_pred) ** 2).mean())

print(eqm)

0.06557438524302006


Code 10 creates two row-matrices: one composed of true values and the other based on predicted values. This is typical of a linear regression analysis.

In [ ]:
# Approach using Dictionary and Pandas Series

a = {1:'b', 2:'c', 3:'c', 4:'b', 5:'a', 6:'a', 7:'d', 8:'d', 9:'a', 10:'a', 11:'a', 12:'b', 13:'d', 14:'b', 15:'b'}

mod_2_answers = pd.Series(data = a, index = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15])

print(mod_2_answers)

1     b
2     c
3     c
4     b
5     a
6     a
7     d
8     d
9     a
10    a
11    a
12    b
13    d
14    b
15    b
dtype: object




---



---



Author: Renan Pereira Bento